packages/client/lib/rpc/modules/eth.ts

```diff
@@ -986,4 +988,53 @@ export class Eth {
     const block = await getBlockByOption(blockOpt, this._chain)
     return intToHex(block.transactions.length)
   }
+
+  /**
+   * Gas price oracle.
+   *
+   * Returns a suggested gas price.
+   * @returns a hex code of an integer representing the suggested gas price in wei.
+   */
+  async gasPrice() {
+    let gasPrice = BigInt(0)
+    const latest = await this._chain.getCanonicalHeadHeader()
```

Comment:
I generally wonder if we want to make this dependent from the `synchronized` status of the client `this.client.config.synchronized`? 🤔 it likely doesn't make sense to return anything (respectively: is not useful) if the client is still syncing and we are not calculating upon the latest blocks?

Not sure, maybe to test this in Geth what is responded in such a case?

Also, then: any reason to not also take the chain common object here for the 1559 check (below)?

Reply:
1. Also, then: any reason to not also take the chain common object here for the 1559 check (below)?

->Made a change where the min gas price from the chain common object is now checked on both (1559 and non 1559) chains.

Reply:
>Not sure, maybe to test this in Geth what is responded in such a case?

I checked this in geth and it does return a gas price even when syncing.  Not sure where that gas price comes from but it's producing something.  The spec doesn't give any specifics about how to calculate current gas price so I guess I would just we go with compute best gas price available based on the current view of the chain that we have (i.e. what is implemented here) and leave it to the consumer to validate the sync status of the client via the `eth_syncing` when evaluating gas price.